File: src/main/java/me/nasukhov/TukitaLearner/bot/io/Channel.java
package me.nasukhov.TukitaLearner.bot.io;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.LocalDateTime;

@Entity
@Table(name = "channels")
public class Channel {
    @Id
    public String id;

    @Column(nullable = false)
    private boolean isPublic;

    @Column(nullable = false)
    private boolean isActive = true;

    @Column(nullable = false)
    private LocalDateTime addedAt;

    protected Channel() {

    }

    public Channel(String id) {
        this.id = id;
        this.isPublic = true;
        addedAt = LocalDateTime.now();
    }

    public Channel(String id, boolean isPublic) {
        this.id = id;
        this.isPublic = isPublic;
        addedAt = LocalDateTime.now();
    }

    public boolean isPublic() {
        return isPublic;
    }

    public boolean isActive() {
        return isActive;
    }

    public void deactivate() {
        isActive = false;
    }

    public void activate() {
        isActive = true;
    }

    public LocalDateTime getRegistrationDate() {
        return addedAt;
    }
}

File: src/main/kotlin/me/nasukhov/tukitalearner/bot/io/Channel.kt
package me.nasukhov.tukitalearner.bot.io

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "channels")
class Channel {
    @Id
    var id: String = ""
        private set

    @Column(nullable = false)
    var isPublic: Boolean = true
        private set

    @Column(nullable = false)
    var isActive: Boolean = true
        private set

    @Column(nullable = false, updatable = false)
    lateinit var addedAt: LocalDateTime
        private set

    // JPA requires a no-arg constructor
    private constructor()

    constructor(id: String) {
        this.id = id
        this.addedAt = LocalDateTime.now()
    }

    constructor(id: String, isPublic: Boolean) {
        this.id = id
        this.isPublic = isPublic
        this.addedAt = LocalDateTime.now()
    }

    fun deactivate() {
        isActive = false
    }

    fun activate() {
        isActive = true
    }
}
